pkg/cmd/browse/browse.go

```diff
@@ -0,0 +1,245 @@
+	cmd.Flags().BoolVarP(&opts.ProjectsFlag, "projects", "p", false, "Open repository projects")
+	cmd.Flags().BoolVarP(&opts.WikiFlag, "wiki", "w", false, "Open repository wiki")
+	cmd.Flags().BoolVarP(&opts.SettingsFlag, "settings", "s", false, "Open repository settings")
+	cmd.Flags().BoolVarP(&opts.BranchFlag, "branch", "b", false, "Select another branch by passing in the branch name")
+
+	return cmd
+}
+
+func openInBrowser(cmd *cobra.Command, opts *BrowseOptions) error {
+
+	baseRepo, err := opts.BaseRepo()
+	httpClient, _ := opts.HttpClient()
+	apiClient := api.NewClientFromHTTP(httpClient)
+	branchName, _ := api.RepoDefaultBranch(apiClient, baseRepo)
+
+	if !inRepo(err) {
+		return printExit(exitNotInRepo, cmd, opts, "")
+	}
+
+	if getFlagAmount(cmd) > 1 {
+		return printExit(exitTooManyFlags, cmd, opts, "")
+	}
+
+	repoUrl := ghrepo.GenerateRepoURL(baseRepo, "")
+	response := exitUrlSuccess
+
+	if !hasArg(opts) && hasFlag(cmd) {
+		response, repoUrl = addFlag(opts, repoUrl)
+	} else if hasArg(opts) && !hasFlag(cmd) {
+		response, repoUrl = addArg(opts, repoUrl, branchName)
+	} else if hasArg(opts) && hasFlag(cmd) {
+		response, repoUrl = addCombined(opts, repoUrl, branchName)
+	}
+
+	if response == exitUrlSuccess || response == exitNonUrlSuccess {
+		opts.Browser.Browse(repoUrl)
+	}
+
+	return printExit(response, cmd, opts, repoUrl)
+}
+
+func addCombined(opts *BrowseOptions, url string, branchName string) (exitCode, string) {
+
+	if !opts.BranchFlag {
+		return exitInvalidCombo, ""
+	}
+
+	if opts.AdditionalArg == "" {
+		return exitUrlSuccess, url + "/tree/" + opts.SelectorArg
+	}
+
+	arr := parseFileArg(opts)
+	if len(arr) > 1 {
+		return exitUrlSuccess, url + "/tree/" + opts.AdditionalArg + "/" + arr[0] + "#L" + arr[1]
+	}
+
+	return exitUrlSuccess, url + "/tree/" + opts.AdditionalArg + "/" + arr[0]
+}
+
+func addFlag(opts *BrowseOptions, url string) (exitCode, string) {
+	if opts.ProjectsFlag {
+		return exitUrlSuccess, url + "/projects"
+	} else if opts.SettingsFlag {
+		return exitUrlSuccess, url + "/settings"
+	} else if opts.WikiFlag {
+		return exitUrlSuccess, url + "/wiki"
+	}
+	return exitExpectedArg, ""
+}
+
+func addArg(opts *BrowseOptions, url string, branchName string) (exitCode, string) {
+
+	if opts.AdditionalArg != "" {
+		return exitTooManyArgs, ""
+	}
+
+	if isNumber(opts.SelectorArg) {
+		url += "/issues/" + opts.SelectorArg
+		return exitNonUrlSuccess, url
+	}
+
+	arr := parseFileArg(opts)
+	if len(arr) > 1 {
+		return exitUrlSuccess, url + "/tree/" + branchName + "/" + arr[0] + "#L" + arr[1]
+	}
+
+	return exitUrlSuccess, url + "/tree/" + branchName + "/" + arr[0]
+}
+
+func parseFileArg(opts *BrowseOptions) []string {
+	arr := strings.Split(opts.SelectorArg, ":")
+	return arr
+}
+
+func printExit(exit exitCode, cmd *cobra.Command, opts *BrowseOptions, url string) error {
+	w := opts.IO.ErrOut
+	cs := opts.IO.ColorScheme()
+	help := "Use 'gh browse --help' for more information about browse\n"
```

Comment:
Instead of including this help in the error message, you can wrap a plain error in a `cmdutil.FlagError`:
```go
return &cmdutil.FlagError{Err: fmt.Errorf("accepts 1 flag, %d flag(s) were recieved", getFlagAmount(cmd))}
```
Now the main program will print the message `accepts 1 flag...` but also print the usage instructions for the current command.